dotCMS/src/main/java/com/dotcms/workflow/helper/WorkflowHelper.java

```diff
@@ -0,0 +1,519 @@
+        newAction.setCommentable(workflowActionForm.isActionCommentable());
+        newAction.setIcon       (workflowActionForm.getActionIcon());
+        newAction.setNextStep   (workflowActionForm.getActionNextStep());
+        newAction.setSchemeId   (workflowActionForm.getSchemeId());
+        newAction.setCondition  (workflowActionForm.getActionCondition());
+        newAction.setRequiresCheckout(workflowActionForm.isRequiresCheckout());
+        newAction.setRequiresCheckoutOption((null != workflowActionForm.getRequiresCheckoutOption())?
+                workflowActionForm.getRequiresCheckoutOption().getName():WorkflowAction.LOCKED_OR_UNLOCKED);
+        newAction.setRoleHierarchyForAssign(workflowActionForm.isRoleHierarchyForAssign());
+
+        try {
+
+            newAction.setNextAssign(this.resolveRole(actionNextAssign).getId());
+            if(!UtilMethods.isSet(newAction.getNextAssign())){
+                newAction.setNextAssign(null);
+            }
+
+            final List<Permission> permissions = new ArrayList<>();
+
+            for (final String permissionName : workflowActionForm.getWhoCanUse()) {
+
+                if (UtilMethods.isSet(permissionName)) {
+
+                    this.processPermission(newAction, permissions, permissionName);
+                }
+            }
+
+            Logger.debug(this, "Saving new Action: " + newAction.getName());
+            this.workflowAPI.saveAction(newAction, permissions);
+
+            if(isNew) {
+
+                // if should be associated to a stepId right now
+                if (UtilMethods.isSet(workflowActionForm.getStepId())) {
+
+                    Logger.debug(this, "The Action: " + newAction.getId() +
+                            ", is going to be associated to the step: " + workflowActionForm.getStepId());
+                    this.workflowAPI.saveAction(newAction.getId(),
+                            workflowActionForm.getStepId(), user);
+                }
+
+                Logger.debug(this, "Saving new WorkflowActionClass, for the Workflow action: "
+                        + newAction.getId());
+
+                addSyncCommitListener(() -> {
+                    WorkflowActionClass workflowActionClass = new WorkflowActionClass();
+                    workflowActionClass.setActionId(newAction.getId());
+                    workflowActionClass.setClazz(NotifyAssigneeActionlet.class.getName());
+                    try {
+                        workflowActionClass.setName(NotifyAssigneeActionlet.class.newInstance().getName());
+                        workflowActionClass.setOrder(0);
+                        this.workflowAPI.saveActionClass(workflowActionClass);
+                    } catch (Exception e) {
+                        Logger.error(this.getClass(), e.getMessage(), e);
+                        throw new DotWorkflowException(e.getMessage(), e);
+                    }
+                });
+            }
+        } catch (Exception e) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)